File: Java/DesignPatterns/src/test/java/designPatterns/creational/prototype/MainPrototypeTest.java
package designPatterns.creational.prototype;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MainPrototypeTest {
  private ConcretePrototype prototype1;
  private ConcretePrototype prototype2;

  @BeforeEach
  void setUp() {
    prototype1 = new ConcretePrototype("initial state");
    prototype2 = (ConcretePrototype) prototype1.clone();
  }

  @Test
  void equalsTest() {
    assertThat(prototype1.showState()).isEqualTo(prototype2.showState());
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/creational/prototype/MainPrototypeTest.kt
package creational.prototype

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MainPrototypeTest {
  private lateinit var prototype1: ConcretePrototype
  private lateinit var prototype2: ConcretePrototype

  @BeforeEach
  internal fun setUp() {
    prototype1 = ConcretePrototype("initial state")
    prototype2 = prototype1.clone() as ConcretePrototype
  }

  @Test
  internal fun equalsTest() {
    assertThat(prototype1.showState()).isEqualTo(prototype2.showState())
  }
}